source/dynamical_systems/src/Circular.cpp

```diff
@@ -55,7 +54,7 @@ void Circular::set_base_frame(const CartesianState& base_frame) {
   }
   IDynamicalSystem<CartesianState>::set_base_frame(base_frame);
   // update reference frame of center
-  if (!this->limit_cycle_->get_value().get_center_state().is_empty()) {
+  if (!this->limit_cycle_->get_value().is_empty()) {
```

Comment:
nitpick: do you prefer this way for readability? I generally prefer `!state.is_empty()` with `state` since the introduction of the bool operator.

```suggestion
  if (this->limit_cycle_->get_value()) {
```

Reply:
No you're right, I was just lazy. I prefer to use either `if (state.is_empty())` or `if (state)`